daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/IdString.scala

```diff
@@ -257,16 +262,16 @@ private[data] final class IdStringImpl extends IdString {
   // We allow space because the navigator's applicationId used it.
   override type LedgerString = String
   override val LedgerString: ConcatenableStringModule[LedgerString, HexString] =
-    new ConcatenableMatchingStringModule("._:-#/ ".contains(_), 255)
+    new ConcatenableMatchingStringModule("DAML LF Ledger String", "._:-#/ ".contains(_), 255)
 
   override type ParticipantId = String
-  override val ParticipantId = LedgerString
+  override val ParticipantId: LedgerString.type = LedgerString
```

Comment:
This adds no value, in fact it contradicts the intended meaning of the interface. IntelliJ is full of it when it comes to these "warnings".